src/OpenTelemetry.Instrumentation.Dependencies/Implementation/HttpWebRequestActivitySource.net461.cs

```diff
@@ -75,35 +76,62 @@ internal sealed class HttpWebRequestDiagnosticSource : DiagnosticListener
+            }
         }
 
-        public override IDisposable Subscribe(IObserver<KeyValuePair<string, object>> observer, Predicate<string> isEnabled)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static void AddRequestTagsAndInstrumentRequest(HttpWebRequest request, Activity activity)
         {
-            IDisposable result = base.Subscribe(observer, isEnabled);
-            this.Initialize();
-            return result;
+            activity.DisplayName = HttpTagHelper.GetOperationNameForHttpMethod(request.Method);
+
+            activity.AddTag(SpanAttributeConstants.ComponentKey, "http");
+            activity.AddTag(SpanAttributeConstants.HttpMethodKey, request.Method);
+            activity.AddTag(SpanAttributeConstants.HttpHostKey, HttpTagHelper.GetHostTagValueFromRequestUri(request.RequestUri));
+            activity.AddTag(SpanAttributeConstants.HttpUrlKey, request.RequestUri.OriginalString);
+            activity.AddTag(SpanAttributeConstants.HttpFlavorKey, HttpTagHelper.GetFlavorTagValueFromProtocolVersion(request.ProtocolVersion));
+
+            InstrumentRequest(request, activity);
+
+            activity.SetCustomProperty("HttpWebRequest.Request", request);
         }
 
-        public override IDisposable Subscribe(IObserver<KeyValuePair<string, object>> observer, Func<string, object, object, bool> isEnabled)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static void AddResponseTags(HttpWebResponse response, Activity activity)
         {
-            IDisposable result = base.Subscribe(observer, isEnabled);
-            this.Initialize();
-            return result;
+            activity.AddTag(SpanAttributeConstants.HttpStatusCodeKey, HttpTagHelper.GetStatusCodeTagValueFromHttpStatusCode(response.StatusCode));
+            activity.AddTag(SpanAttributeConstants.HttpStatusTextKey, response.StatusDescription);
+
+            activity.SetCustomProperty("HttpWebRequest.Response", response);
         }
 
-        public override IDisposable Subscribe(IObserver<KeyValuePair<string, object>> observer)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static void AddExceptionTags(Exception exception, Activity activity)
         {
-            IDisposable result = base.Subscribe(observer);
-            this.Initialize();
-            return result;
+            if (exception is WebException wexc && wexc.Response is HttpWebResponse response)
+            {
+                activity.AddTag(SpanAttributeConstants.HttpStatusCodeKey, HttpTagHelper.GetStatusCodeTagValueFromHttpStatusCode(response.StatusCode));
+                activity.AddTag(SpanAttributeConstants.HttpStatusTextKey, response.StatusDescription);
+            }
+            else
+            {
+                activity.AddTag("error", exception.Message);
+            }
+
+            activity.SetCustomProperty("HttpWebRequest.Exception", exception);
```

Comment:
if some processor/exporter wants to add exception.Message, then can do so by reading this exception. That means we can avoid adding the "error" tag.